keps/sig-windows/3636-windows-csi-host-process-pods/README.md

```diff
@@ -0,0 +1,1248 @@
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+CSI Proxy enabled running CSI in Windows nodes using a client/server model. The server is the CSI Proxy binary
+running as a Windows service in the node, and the client is the CSI Driver, which communicates with CSI Proxy on
+every CSI call done on the node plugin. While this model has worked fine, it has a few drawbacks:
+
+- Different deployment model than Linux - Linux privileged containers are used to perform the privileged storage
+  operations (format/mount). However, Windows containers aren't privileged. To work around the problem, the CSI Driver is run as non-privileged containers,
+  and privileged operations are relayed to CSI Proxy. In deployment manifests, the Windows component needs to have an
+  additional section to mount the named pipes exposed by CSI Proxy as a hostpath.
+- Additional component in the host to maintain - The cluster administrator needs to install and run CSI Proxy
+  during the node bootstrap. The cluster administrator also needs to think about the upgrade workflow in addition to
+  upgrading the CSI Driver.
+- Difficult releases of bugfixes & features - After a bugfix, we create a new version of the CSI Proxy to be
+  redeployed in the cluster. After a feature is merged, in addition to redeploying a new version of CSI Proxy,
+  the client needs to be updated with a new version of the CSI Proxy client and connect to the new version of the named pipes,
+  this workflow is not as simple as the Linux counterpart which only needs to update Go dependencies.
+- Multiple API versions to maintain - As part of the original design of CSI Proxy it was decided to have different
+  protobuf versions whenever there were breaking changes (like updates in the protobuf services & messages), this lead
+  to having multiple versions of the API (v1alphaX, v1betaX, v1). In addition, if we want to add a new feature we'd need
+  to create a new API version e.g. v2alpha1 ([see this PR as an example of adding methods to the Volume API Group](https://github.com/kubernetes-csi/csi-proxy/pull/186)).
+
+In 1.22, SIG Windows introduced [HostProcess containers](https://kubernetes.io/blog/2021/08/16/windows-hostprocess-containers/)
+as an alternative way to run containers. HostProcess containers run directly in the host
+and behave like to a regular process.
+
+Using HostProcess containers in CSI Drivers enables CSI Drivers to perform the privileged storage operations
+directly. Most of the drawbacks in the client/server model are no longer present in the new model.
+
+### Goals
```

Comment:
Another goal: Identify node version skew problems, this is in the CSI Proxy guide but should also be in the KEP, ref https://github.com/kubernetes-csi/csi-proxy/pull/285